bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,913 @@
+            @Nullable VerisureInstallation installation) {
+        if (installation != null) {
+            BigDecimal installationId = installation.getInstallationId();
+            String url = START_GRAPHQL;
+
+            String queryQLClimates = "{\"operationName\":\"Climate\",\"variables\":{\"giid\":\"" + installationId
+                    + "\"},\"query\":\"query Climate($giid: String!) {\\n installation(giid: $giid) {\\n climates {\\n device {\\n deviceLabel\\n area\\n gui {\\n label\\n __typename\\n }\\n __typename\\n }\\n humidityEnabled\\n humidityTimestamp\\n humidityValue\\n temperatureTimestamp\\n temperatureValue\\n __typename\\n }\\n __typename\\n }\\n}\\n\"}";
+            logger.debug("Trying to get climate status with URL {} and data {}", url, queryQLClimates);
+            VerisureClimatesJSON thing = (VerisureClimatesJSON) callJSONRestPost(url, queryQLClimates, jsonClass);
+            logger.debug("REST Response ({})", thing);
+
+            if (thing != null && thing.getData() != null) {
+                List<VerisureClimatesJSON.Climate> climateList = thing.getData().getInstallation().getClimates();
+                for (VerisureClimatesJSON.Climate climate : climateList) {
+                    VerisureClimatesJSON cThing = new VerisureClimatesJSON();
+                    VerisureClimatesJSON.Installation inst = new VerisureClimatesJSON.Installation();
+                    List<VerisureClimatesJSON.Climate> list = new ArrayList<VerisureClimatesJSON.Climate>();
+                    list.add(climate);
+                    inst.setClimates(list);
+                    VerisureClimatesJSON.Data data = new VerisureClimatesJSON.Data();
+                    data.setInstallation(inst);
+                    cThing.setData(data);
+                    // Set unique deviceID
+                    String deviceId = climate.getDevice().getDeviceLabel();
+                    if (deviceId != null) {
+                        cThing.setDeviceId(deviceId);
+                        // Set location
+                        cThing.setLocation(climate.getDevice().getArea());
+                        VerisureThingJSON oldObj = verisureThings.get(cThing.getDeviceId());
+                        if (oldObj == null || !oldObj.equals(cThing)) {
+                            cThing.setSiteId(installation.getInstallationId());
+                            cThing.setSiteName(installation.getInstallationName());
+                            verisureThings.put(cThing.getDeviceId(), cThing);
+                            notifyListeners(cThing);
+                        }
+                    }
+                }
+            } else {
+                logger.warn("Failed to update ClimateStatus thing: {}, thing.getData: {}", thing, thing.getData());
+            }
+        }
+    }
+
+    private synchronized void updateDoorWindowStatus(Class<? extends VerisureThingJSON> jsonClass,
```

Comment:
These status updates methods in this class seems to have lot of duplicate code. Could they be simplified for sake of maintainability?

Reply:
I've not found any major simplification since they handle completely different Verisure things, but I will continue to think about it.